packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+      }
+    },
+    [getItemRef, listRef, stickyItemIds, visibleItemIds],
+  );
+
+  // When using keyboard, and navigate to stickyItems, arrow up/down should navigate to previous item's last child/current Item's first child.
+  // But because of virtualization, the destination item is not always rendered, so we scroll to them to force rendering
+  const handleArrowUpDownOnSticky = React.useCallback(
+    (stickyId, stickyItem) => (e: React.KeyboardEvent) => {
+      if (e.key === 'ArrowDown') {
+        e.preventDefault();
+        focusItemById(stickyItem.childrenIds[0]);
+      }
+      if (e.key === 'ArrowUp' && stickyItem.index !== 1) {
+        const indexAmongVisible = visibleItemIds.indexOf(stickyId);
+        if (indexAmongVisible > 0) {
+          e.preventDefault();
+          focusItemById(visibleItemIds[indexAmongVisible - 1]);
+        }
+      }
+    },
+    [focusItemById, visibleItemIds],
+  );
+
+  const createTreeItem = React.useCallback(
+    (id, style) => {
+      const item = getItemById(id);
+      if (item) {
+        const { expanded, parent, level, index, treeSize, childrenIds } = item;
+        return TreeItem.create(item.item, {
+          defaultProps: () =>
+            getA11yProps('item', {
+              renderItemTitle: props.renderItemTitle,
+            }),
+          overrideProps: {
+            style, // came from react-window
+            expanded,
+            parent,
+            key: id,
+            level,
+            index,
+            treeSize,
+            selectable: false,
+            onFocus: () => makeVisibleOnFocus(id, level),
+            ...(level === 1 && expanded && childrenIds.length && { onKeyDown: handleArrowUpDownOnSticky(id, item) }),
+          },
+        });
+      }
+      return null;
+    },
+    [getA11yProps, getItemById, handleArrowUpDownOnSticky, makeVisibleOnFocus, props.renderItemTitle],
+  );
+
+  const innerElementContextValue: InnerElementContextType = React.useMemo(
+    () => ({
+      getItemById,
+      stickyItemIds,
+      stickyItemPusherHeights,
+      stickyItemSize,
+    }),
+    [getItemById, stickyItemIds, stickyItemPusherHeights, stickyItemSize],
+  );
+
+  const getItemKey = React.useCallback((index: number, data: VirtualItemData) => data.visibleItemIds[index], []);
+
+  const getItemSize = React.useCallback(
+    (index: number) => {
+      const id = visibleItemIds[index];
+
+      const item = getItemById(id);
+      if (item?.level === 1) {
+        return stickyItemSize;
+      }
+      return itemSize;
+    },
+    [getItemById, itemSize, stickyItemSize, visibleItemIds],
+  );
+
+  React.useLayoutEffect(() => {
+    listRef.current.resetAfterIndex(0);
+  }, [listRef, visibleItemIds]); // when item collapsed/expanded (visibleItemIds change), refresh react-window itemSize cache
+
+  const element = (
+    <TreeContext.Provider value={contextValue}>
+      {getA11yProps.unstable_wrapWithFocusZone(
```

Comment:
just curious, what does this do ?

Reply:
It wraps our component in `<FocusZone`, and provide a bunch of keyboard navigation.
I don't know 100% percent of it. I know one reason this is important, without it, pressing down arrow key will not be scrolling the list